app/controllers/spree/api/v2/display_products_controller.rb

```diff
@@ -5,9 +5,9 @@ module Api
     module V2
       class DisplayProductsController < Spree::Api::V2::ResourceController
         include Spree::Api::V2::CollectionOptionsHelpers
+        before_action :require_spree_current_user
```

Comment:
@mausamp we do not need logged in user to list index.. you should not have changed this